--- gimp-2.10.12/configure.ac.orig
+++ gimp-2.10.12/configure.ac
@@ -2124,8 +2124,8 @@
   dinput_save_CPPFLAGS="$CPPFLAGS"
   dinput_save_LDFLAGS="$LDFLAGS"
   if test "x$with_directx_sdk" != x; then
-    CPPFLAGS="-I $with_directx_sdk/Include $CPPFLAGS"
-    LDFLAGS="-L$with_directx_sdk/Lib/x86 $LDFLAGS"
+    CPPFLAGS="-I $with_directx_sdk/include $CPPFLAGS"
+    LDFLAGS="-L$with_directx_sdk/lib $LDFLAGS"
     AC_CHECK_HEADER(dinput.h,
                     [AC_CHECK_LIB(dxguid, GUID_XAxis, have_dx_dinput=yes)])
   fi
